Activities/ManagePersonsPermissionCheckerTask.cs

```diff
@@ -1,74 +1,75 @@
         IUserService userService,
         IWorkflowExpressionEvaluator expressionEvaluator,
-        IRoleService roleService,
         IStringLocalizer<ManagePersonsPermissionCheckerTask> localizer)
     {
+        _authorizationService = authorizationService;
+        _userService = userService;
         _expressionEvaluator = expressionEvaluator;
         S = localizer;
-        _userService = userService;
-        _roleService = roleService;
     }
 
+    // The technical name of the activity. Activities on a workflow definition reference this name.
     public override string Name => nameof(ManagePersonsPermissionCheckerTask);
 
+    // The displayed name of the activity, so it can use localization.
     public override LocalizedString DisplayText => S["Manage Persons Permission Checker Task"];
 
+    // The category to which this activity belongs. The activity picker groups activities by this category.
     public override LocalizedString Category => S["User"];
+
+    // The username to evaluate for ManagePersons permission.
     public WorkflowExpression<string> UserName
     {
         get => GetProperty(() => new WorkflowExpression<string>());
         set => SetProperty(value);
     }
 
+    // Returns the possible outcomes of this activity.
     public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext) =>
         Outcomes(S["HasPermission"], S["NoPermission"]);
 
+    // This is the heart of the activity and actually performs the work to be done.
     public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
     {
-        var userName = await _expressionEvaluator.EvaluateAsync(UserName, workflowContext, null);
-        User user = (User)await _userService.GetUserAsync(userName);
+        var userName = await _expressionEvaluator.EvaluateAsync(UserName, workflowContext, encoder: null);
+        var user = (User)await _userService.GetUserAsync(userName);
 
         if (user != null)
         {
-            var permissions = user.RoleNames
-                .Select(async roleName => await _roleService.GetRoleClaimsAsync(roleName))
-                .SelectMany(claim => claim.Result)
-                .Where(claim => "Permission".Equals(claim.Type))
-                .Select(claim => claim.Value)
-                .Distinct()
-                .ToList();
+            var userClaim = await _userService.CreatePrincipalAsync(user);
 
-            return permissions.Contains("ManagePersons") ? Outcomes("HasPermission") : Outcomes("NoPermission");
+            if (await _authorizationService.AuthorizeAsync(userClaim, PersonPermissions.ManagePersons))
+            {
+                return Outcomes("HasPermission");
+            }
         }
 
         return Outcomes("NoPermission");
     }
 }
+
+// NEXT STATION: ViewModels/ManagePersonsPermissionCheckerTaskViewModel.cs
```

Comment:
Follow these messages throughout to "END OF TRAINING SECTION". All files should be visited in the end.